classes/module/Module.php

```diff
@@ -2441,6 +2441,9 @@ public function display($file, $template, $cache_id = null, $compile_id = null)
             if ($cache_id !== null) {
                 Tools::enableCache();
             }
+            if ($compile_id === null) {
+                $compile_id = Context::getContext()->shop->theme->getName();
```

Comment:
unfortunately, the name is not uniq :thinking: 
Why not using the shop_id instead?

Reply:
What would happen then in multistore, if we change the theme for this shop, will the cache automatically be updated for this cache id ?

Reply:
Did you check the existing fetch function in Module.php? It uses this logic and is missing in other functions which have compile_id.